exir/backend/canonical_partitioners/group_partitioner.py

```diff
@@ -133,97 +133,60 @@ def _process_node_groups(
         partition_users,
         partition_map,
     ):
-        """Process nodes in predefined groups."""
-        group_to_partition_id = {}
-
-        if not self.node_groups:
-            return group_to_partition_id
-
-        processed_nodes = set()
-
-        # We have to create the partitions in reverse topological order
-        # so we find the groups as we traverse backwards in the graph
-        # this likely needs to be combined with the process_remaining_nodes
-        # TODO: this currently doesn't work with _process_remaining_nodes so
-        # if a user provides grouped nodes with operatorsupport, then this will
-        # faile
+        """Process nodes not in any predefined group."""
         for node in reversed(self.graph_module.graph.nodes):
-            if node not in self.node_to_group:
+            if node in assignment or not self._is_node_supported(node):
                 continue
 
-            if node in processed_nodes:
```

Comment:
what about this check? The purpose was that when traversing nodes in the graph module, we could get two nodes from the same group, and we could potentially add the group twice?